### DISCUSSION TITLE
Allowing for the expo-router way?

### DISCUSSION THREAD

**Discussion by pvinis** [react-navigation/react-navigation#11087] [Ideas]
expo-router has been going a different way with navigation. Many people (including me) feel like it's a good way to do navigation, but I know that it might not be the same for everyone.

Has there been any talks about moving react-navigation to a model like the one in expo-router? Or maybe a fork of react-nav, or even better, a way to support both old-school style, and expo-router style?

For reference, this is what I'm talking about:
- https://expo.github.io/router/docs/features/dynamic-routes
- https://expo.github.io/router/docs/features/routes

**Comment by satya164** [comment]
Are you referring to file system-based routing? If yes then that's not something React Navigation can do. React Navigation is a library that you use, not a framework such as Next.js which has control of the tooling. FS-based routing requires deeper integration with the bundler that React Navigation doesn't have as a library.

But other frameworks can use React Navigation to implement such a pattern - which is what the Expo Router does. I don't think React Navigation needs to compete with Expo Router for this - we can focus on providing the building blocks and primitives required for advanced use cases and frameworks such as Expo Router can make use of those APIs to implement the pattern they want to. And consumers have the choice to use React Navigation directly or use it through a framework such as Expo Router depending on what they prefer.

**Reply by pvinis** [reply]
gotcha! thank you, makes sense. maybe expo router would be available for non expo projects later. or some fork of it.

thank you again.
